### PR TITLE
Make it possible to enable encryption at host for a VM

### PR DESCRIPTION
**Pull request by freva**
# Description

This adds a method to enable Encryption at Host when provisioning a VM.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by freva** [comment]
@microsoft-github-policy-service agree [company="Vespa.ai"]

**Comment by freva** [comment]
@microsoft-github-policy-service agree company="Vespa.ai"

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java [hunk 1]
```diff
@@ -2916,6 +2916,12 @@ private UefiSettings ensureUefiSettings() {
         return uefiSettings;
     }
 
+    @Override
+    public VirtualMachine.DefinitionStages.WithCreate withEncryptionAtHost() {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:2920]: Here it should return `VirtualMachineImpl`.

**Review by weidongxu-microsoft — COMMENTED** [review]
@XiaofeiCao We need to ensure the feature is no long in preview.

So far for official doc I no longer see any mention of "preview" now.
https://learn.microsoft.com/en-us/azure/virtual-machines/disks-enable-host-based-encryption-portal?tabs=azure-powershell

**Comment by XiaofeiCao** [comment]
@weidongxu-microsoft Yeah, it's GAed:
https://azure.microsoft.com/en-us/updates/general-availability-encryption-at-host-for-premium-ssd-v2-and-ultra-disks/

@freva Thanks for the contribution. We'll probably need an update option as well. Let me add those in this PR.

**Review by XiaofeiCao — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by XiaofeiCao** [comment]
Hi @freva , 2.35.0 has released with encryption at host support.

**Comment by freva** [comment]
Thanks, @XiaofeiCao! :tada: :rocket:

**Pull request merged**